### DISCUSSION TITLE
如何在ts中查找DataTable的某一行？

### DISCUSSION THREAD

**Discussion by NashXY** [Tencent/puerts#1859] [Q&A] [answered]
https://github.com/Tencent/puerts/issues/986
这个问题我看过了，照抄它的代码只能获得Struct的默认值

`let DataTable = UE.DataTable.Load("DataTable'/Game/StarterContent/DT_Test.DT_Test'")
let TestStruct1 = UE.UserDefinedStruct.Load("UserDefinedStruct'/Game/StarterContent/TestStruct.TestStruct'");
let Test1 = UE.NewStruct(TestStruct1);

let Test2 = Test1 as UE.TableRowBase;
UE.DataTableFunctionLibrary.GetDataTableRowFromName(DataTable, "1", $ref(Test2))

console.log(`struct /////////////${(Test1 as UE.Game.StarterContent.TestStruct.TestStruct).age}`)
console.log(`struct /////////////${(Test1 as UE.Game.StarterContent.TestStruct.TestStruct).speed}`)
console.log(`struct /////////////${(Test1 as UE.Game.StarterContent.TestStruct.TestStruct).name}`)

UE.DataTableFunctionLibrary.GetDataTableRowFromName(DataTable, "2", $ref(Test2))

console.log(`struct /////////////${(Test1 as UE.Game.StarterContent.TestStruct.TestStruct).age}`)
console.log(`struct /////////////${(Test1 as UE.Game.StarterContent.TestStruct.TestStruct).speed}`)
console.log(`struct /////////////${(Test1 as UE.Game.StarterContent.TestStruct.TestStruct).name}`)
//`

请教一下有没有其他方法能够成功在Ts中查到DataTable数据？

**Comment by chexiongsheng** [comment]
你认真看issue了吗？这issue就是反馈你这写法崩溃了啊。然后你照抄崩溃写法？
issue后面不是有我写的正确写法？？

**Reply by NashXY** [reply, marked as answer]
不好意思没注意，这样改后确实可以

`
        let TestStruct1 = UE.UserDefinedStruct.Load("UserDefinedStruct'/Game/Blueprints/Type/Struct/Weapon/GunInfo.GunInfo'");
        let Test1 = UE.NewStruct(TestStruct1);
        UE.DataTableFunctionLibrary.Generic_GetDataTableRowFromName(this.GunDataTable, "1001", Test1);
        const result = Test1 as UE.Game.Blueprints.Type.Struct.Weapon.GunInfo.GunInfo;
`
